src/sentry/reprocessing2.py

```diff
@@ -0,0 +1,178 @@
+    # under a new event ID. The second step happens in pre-process. We could
+    # save the "original event ID" instead and get away with writing less to
+    # nodestore, but doing it this way makes the logic slightly simpler.
+
+    # Step 1: Fix up the event payload for reprocessing and put it in event
+    # cache/event_processing_store
+    tags = dict(data.get("tags") or ())
+    orig_event_id = tags["original_event_id"] = data["event_id"]
+    event = eventstore.get_event_by_id(project_id, orig_event_id)
+    if event is None:
+        return
+
+    tags["original_group_id"] = event.group_id
+    data["tags"] = list(tags.items())
+
+    event_id = data["event_id"] = uuid.uuid4().hex
+
+    # XXX: Only reset received
+    data["timestamp"] = data["received"] = start_time
+
+    cache_key = event_processing_store.store(data)
+
+    # Step 2: Copy attachments into attachment cache
+    queryset = EventAttachment.objects.filter(
+        project_id=project_id, event_id=orig_event_id
+    ).select_related("file")
+
+    attachment_objects = []
+
+    for attachment_id, attachment in enumerate(queryset):
+        with sentry_sdk.start_span(op="reprocess_event.copy_attachment") as span:
+            span.set_data("attachment_id", attachment.id)
+
+            fp = attachment.file.getfile()
+            chunk = None
+            chunk_index = 0
+            size = 0
+            while True:
+                chunk = fp.read(settings.SENTRY_REPROCESSING_ATTACHMENT_CHUNK_SIZE)
+                if not chunk:
+                    break
+
+                size += len(chunk)
+
+                attachment_cache.set_chunk(
+                    key=cache_key,
+                    id=attachment_id,
+                    chunk_index=chunk_index,
+                    chunk_data=chunk,
+                    timeout=CACHE_TIMEOUT,
+                )
+                chunk_index += 1
+
+            assert size == attachment.file.size
+
+            attachment_objects.append(
+                CachedAttachment(
+                    key=cache_key,
+                    id=attachment_id,
+                    name=attachment.name,
+                    # XXX: Not part of eventattachment model, but not strictly
+                    # necessary for processing
+                    content_type=None,
+                    type=attachment.file.type,
+                    chunks=chunk_index,
+                    size=size,
+                )
+            )
+
+    if attachment_objects:
+        with sentry_sdk.start_span(op="reprocess_event.set_attachment_meta"):
+            attachment_cache.set(cache_key, attachments=attachment_objects, timeout=CACHE_TIMEOUT)
+
+    preprocess_event_from_reprocessing(
+        cache_key=cache_key, start_time=start_time, event_id=event_id
+    )
+
+
+def is_reprocessed_event(data):
+    return bool(_get_original_event_id(data))
+
+
+def _get_original_event_id(data):
+    return dict(data.get("tags") or ()).get("original_event_id")
```

Comment:
_nit_: The event model has a `get_tag` function, which we might want to consider to factor out and reuse here. It seems that in most cases you have an event model already, so maybe you can just call `get_tag` directly?

Reply:
in `create_failed_event` I have no event model unfortunately